tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -375,10 +375,11 @@ def stack_new_level_core(self, X, y, models: Union[List[AbstractModel], dict], X
                     for model_name in model_args_fit if 'hyperparameter_tune_kwargs' in model_args_fit[model_name]
                 }
                 kwargs['hyperparameter_tune_kwargs'] = hyperparameter_tune_kwargs
-            if ag_args and 'fit_with_prune' in ag_args:
-                kwargs['fit_with_prune'] = ag_args['fit_with_prune']
-            else:
-                kwargs['fit_with_prune'] = None
+                fit_with_prune_kwargs = {
+                    model_name: model_args_fit[model_name]['fit_with_prune']
```

Comment:
Why does 'fit_with_prune' suddenly become 'fit_with_prune_kwargs'? It should be one or the other, not both.

Reply:
I'll change everything to `fit_with_prune_kwargs`. I made that change because having kwargs at the end made it more clear that it is a dictionary not some boolean flag.